### PR TITLE
Initial Pcap-NG support

### PR DESCRIPTION
**Pull request by p-l-**
We might want to merge `rdpcap` & `rdpcapng`, `PcapReader` & `PcapNgReader`, `RawPcapReader` & `RawPcapNgReader`; but at least this gives an implementation for Pcap-NG file format.

### CODE REVIEW

File: scapy/utils.py [hunk 1]
```diff
@@ -542,6 +542,17 @@ def rdpcap(filename, count=-1):
     fdesc.close()
     return result
 
+def rdpcapng(filename, count=-1):
+    """Read a pcap file and return a packet list
```
Comment (guedou) [inline scapy/utils.py]: s/pcap/pcap-ng/

File: scapy/utils.py [hunk 2]
```diff
@@ -664,7 +676,104 @@ def read_all(self,count=-1):
         return plist.PacketList(res,name = os.path.basename(self.filename))
     def recv(self, size=MTU):
         return self.read_packet(size=size)
-        
+
+
+class RawPcapNgReader(RawPcapReader):
+    """A stateful pcapng reader. Each packet is returned as a
```
Comment (guedou) [inline scapy/utils.py]: pcapng or pcap-ng ? #pickyreviewer

Comment (p-l-) [inline scapy/utils.py]: According to [Wireshark Wiki](https://wiki.wireshark.org/Development/PcapNg), "PCAP Next Generation Dump File Format (or pcapng for short)" (#luckycoder ;-)).

File: scapy/utils.py [hunk 3]
```diff
@@ -664,7 +676,104 @@ def read_all(self,count=-1):
...
+            1: self.read_block_idb,
+            6: self.read_block_epb,
+        }
+        try:
+            self.f = gzip.open(filename,"rb")
+            magic = self.f.read(4)
+        except IOError:
+            self.f = open(filename,"rb")
+            magic = self.f.read(4)
+        if magic != "\n\r\r\n": # PcapNg:
```
Comment (guedou) [inline scapy/utils.py]: I guess that \x0a\x0d\x0d\x0a will be better to stick with the specs. Also renaming it block_type might ease future modifications.

**Comment by guedou** [comment]
Thanks for this cool patch ! I think that it is better to keep pcap & pcapng separated. We might want rdpcap() to be smarter and able to detect pcapng.

I wonder if you could add a simple UT that will test RawPcapNgReader.

File: scapy/utils.py (hunk 3 above)
Comment (p-l-) [inline scapy/utils.py]: Agreed.

**Comment by p-l-** [comment]
@guedou thanks for the review, I'll work on that.

**Comment by p-l-** [comment]
@guedou I think I have handled all your comments, thanks!

**Pull request merged**